File: 0872-leaf-similar-trees/0872-leaf-similar-trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */
function countLeaf(root,leaf)
{
    if (!root) return;
    if (!root.right  && !root.left )
    {
        leaf.push(root.val);
        return;
    }
    countLeaf(root.right,leaf);
    countLeaf(root.left,leaf);
}
var leafSimilar = function(root1, root2) {
    var leaf1 = [];
    var leaf2 = [];
    countLeaf(root1,leaf1);
    countLeaf(root2,leaf2);
     for(let i = 0; i < leaf1.length && i < leaf2.length ;i++)
        if (leaf1[i] != leaf2[i])
            return false;
    return leaf1.length == leaf2.length;

};

File: 0872-leaf-similar-trees/0872-leaf-similar-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void countleaf(TreeNode* root,vector<int> &leaf)
    {
        if (!root) return;
        if (!root->right && !root->left)
        {
            leaf.push_back(root->val);
            return;
        }
        countleaf(root->right,leaf);
        countleaf(root->left,leaf);
    }
    bool leafSimilar(TreeNode* root1, TreeNode* root2) {
        vector<int> tree1;
        vector<int> tree2;
        countleaf(root1,tree1);
        countleaf(root2,tree2);
        return (tree1 == tree2);
    }
};